docs/specs/lcb-packages.md

```diff
@@ -0,0 +1,246 @@
+# LiveCode Builder Packages
+Copyright 2015 LiveCode Ltd.
+
+## Introduction
+
+LiveCode Builder is built around the idea of 'modules' - independently compiled
+units of code which can be selectively used by others.
+
+In a larger system, however, just the compiled module code is not enough. There
+are other things which are naturally related to the module which don't make sense
+to be included within the modules binary file - principally because they are not
+of direct interest to the virtual machine which must execute the code.
+
+To solve this problem, compiled LCB modules are embedded in a more general
+container called a 'package'. A package is a ZIP format archive file with a
+standard structure which combines modules together with other related resources
+and metadata which can be used by an environment loading it.
+
+## Structure
+
+An LCB package is a ZIP archive with a standard layout:
+
+    <root>/

```

Comment:
Do you mean the `<root>` will be a specified folder name, as you seem to suggest below (`com.example.foo.lcp/`), or just that this is the root of the archive?  Just looking at an odt file on my machine and they seems to have everything in the root of the archive.